libraries/app/database_api.cpp

```diff
@@ -624,6 +625,89 @@ vector<optional<account_object>> database_api_impl::get_accounts(const vector<ac
    return result;
 }
 
+vector<limit_order_object> database_api::get_account_limit_orders( const string& name_or_id, const string &base, const string &quote, uint32_t limit, optional<limit_order_id_type> ostart_id, optional<price> ostart_price)
+{
+   return my->get_account_limit_orders( name_or_id, base, quote, limit, ostart_id, ostart_price );
+}
+
+vector<limit_order_object> database_api_impl::get_account_limit_orders( const string& name_or_id, const string &base, const string &quote, uint32_t limit, optional<limit_order_id_type> ostart_id, optional<price> ostart_price)
+{
+   FC_ASSERT( limit <= 100 );
+
+   vector<limit_order_object>   results;
+   const account_object*        account = nullptr;
+   uint32_t                     count = 0;
+   //bool                         v_id = false;       // is start_id valid?
+   //bool                         v_price = false;    // is start_price valid?
+   limit_order_id_type          start_id;
+   price                        start_price;
+
+   if (std::isdigit(name_or_id[0]))
+      account = _db.find(fc::variant(name_or_id, 1).as<account_id_type>(1));
+   else
+   {
+       const auto& idx = _db.get_index_type<account_index>().indices().get<by_name>();
+       auto itr = idx.find(name_or_id);
+       if (itr != idx.end())
+           account = &*itr;
+   }
+   if (account == nullptr)
+       return results;
+
+   auto assets = lookup_asset_symbols( {base, quote} );
+   FC_ASSERT( assets[0], "Invalid base asset symbol: ${s}", ("s",base) );
+   FC_ASSERT( assets[1], "Invalid quote asset symbol: ${s}", ("s",quote) );
+
+   auto base_id = assets[0]->id;
+   auto quote_id = assets[1]->id;
```

Comment:
Here is the best place to check whether the start price is consistent with `base` and `quote` parameters. If not, IMHO best to throw out an exception.